### ISSUE TITLE
clean up USDA data so we can merge into EGRET

### ISSUE THREAD

**Issue by lizzieinvancouver** [lizzieinvancouver/egret#53]
@dbuona and @FrederikBaumgarten were working on this. @FrederikBaumgarten will post code to hopefully do this and flag for @lizzieinvancouver to look at. BTW, code that may be useful (but I am not sure how you do it so I made need to adapt ... we can discuss):

```
# I make imaginary data for example
df <- data.frame(a=c(0,0,1,1), b=c(1,1,1,0), c=c(0,1,0,1), d=c(1,1,0,1))
# get all unique combinations 
dftreatments <- expand.grid(unique(df$a), unique(df$b), unique(df$c), unique(df$d))
dftreatments$howmany <- NA
for(i in c(1:nrow(dftreatments))){
	subby <- df[which(df$a==dftreatments$Var1[i] & 
		df$b==dftreatments$Var2[i] & 
		df$c==dftreatments$Var3[i] & 
		df$d==dftreatments$Var4[i]),]
	dftreatments$howmany[i] <- nrow(subby)
}

```

**Comment by dbuona** [comment]
@dbuona check in with @ngoj1 about how this is going

**Comment by FrederikBaumgarten** [comment]
@dbuona: I added [cleaningUSDA_forDan.r](https://github.com/lizzieinvancouver/egret/blob/main/analyses/cleaningUSDA_forDan.r). The script does 2 things:
1. basic cleaning/merging of similar variables
2. splitting entries with min/max values of chilling
You can merge the code to yours as you like

**Comment by FrederikBaumgarten** [comment]
@lizzieinvancouver : I implemented your code suggestions to calculate treatment combinations for usda [here](https://github.com/lizzieinvancouver/egret/blob/main/analyses/treatComboUSDA.R), Lines 185 - 222. This doesn't really belong in this issue, but there is no other one for it.

**Comment by dbuona** [comment]
@FrederikBaumgarten @ngoj1 Okay ya'll,

 I am trying to merge Justin and Fredi's usda cleaning Scripts(, cleaningUSDA_forDan.r) by sourcing them sequentially in cleanmerge_all_usda.R. 

Fredi based his cleaning on the data file usdaGerminationData.csv which is supposed to be roughly the same as the state of the data at the end of cleanmerge_all_usda_JNVER.R, only it's not.

Fredi's code uses a column called chill.duration at line 32, which apparently exists in usdaGerminationData.csv but not in Justin's new cleaning code. @ngoj1  can you help me figure out where this column was made previously and add it back into
cleanmerge_all_usda_JNVER.R or tell fredi and I which other column represents it?

**Comment by ngoj1** [comment]
> @ngoj1 can you help me figure out where this column was made previously and add it back into
cleanmerge_all_usda_JNVER.R or tell fredi and I which other column represents it?

In my script, I retained chill.duration as a column when moving things over to unify chilling and cold stratification into one column. When I check over Fredi's code, I didn't see chill.duration in line 32 but rather "chill.duraton" in line 34, which is probably because when I first submitted my script I made a spelling error and accidentally omitted in "i" in "duration" which was fixed at a later date. @FrederikBaumgarten, can I change this to spell out "chill.duration" instead, and then we see if the code works afterward?

**Comment by FrederikBaumgarten** [comment]
@ngoj1 : not sure if I get your point right. So you first ommitted the "i" in "duration" but if you fix this in your script then we end up with the corrected column in usdaGerminationData.csv ? That sounds like an easy fix then!

**Comment by ngoj1** [comment]
@FrederikBaumgarten Maybe I'm also confused actually...a column called "chill.duration" does exist in my script and the final output, though the columns that actually hold the consolidated data are called chill.dur.Avg.comb (and Min or Max instead of Avg) since I broke up the ranges in the original data into separate values. 

I've changed the "chill.duraton" to "chill.duration" in Fredi's cleanmerge_all_usda_forDan.R script. When I run line 18 in this script before running line 30 in cleanmerge_all_usda.R, I no longer get a warning about chill.duration not existing, though I do get an error pertaining to line 81 as follows:
```
Error in `usda[!usda$X %in% tmp_X, ]` at scrapeUSDAseedmanual/cleaning/cleaningUSDA_forDan.r:81:0:
! Can't subset rows with `i`.
✖ Logical subscript `i` must be size 1 or 6180, not 0.
Run `rlang::last_trace()` to see where the error occurred.
Warning messages:
1: Unknown or uninitialised column: `X`. 
2: Unknown or uninitialised column: `X`.
```
And if I **don't** run line 18 in cleanmerge_all_usda_forDan.R before running line 30 in cleanmerge_all_usda.R, I get a warning that object "usda.orig" is not found. So maybe the chill.duration dilemma alone is fixed? and these warnings are just to do with some other problem in trying to combine these two scripts together?
